viewing_party/party.py

```diff
@@ -1,23 +1,212 @@
+        if item not in friends_movies: # if the movie is not in the friend's movie list
+            unique.append(item) # we add it to the list of the user_data["watched"]
+    return unique
+
+ # part 2   
+def get_friends_unique_watched(user_data):
+    # will need to access watched inside user_data
+    user_movies= []
+    friends_movies = []
+    # create a new list left_to_watch
+    left_to_watch = []
+    
+    # and compare it to friends["watched"]
+    for movie in user_data["watched"]:
+        user_movies.append(movie) #user_movies has a list of the movies the user has seen
+
+    for movie in user_data["friends"]: # reaching into the list of dictionaries
+        for item in movie["watched"]:
+            friends_movies.append(item) # this is now a list of the friend's movies
+
+    for item in friends_movies: # iterate through the list of friend's movies
+        if item not in user_movies: # if movie in friends["watched"] is not in user_data["watched"]
+            if item not in left_to_watch:
+                left_to_watch.append(item)
+    return left_to_watch
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    user_movies = []
+    friends_movies = []
+    unique = []
+
+    for movie in user_data["watched"]: # this is to help create the user watch list
+        user_movies.append(movie) #user_movies has a list of the movies the user has seen
+
+    for movie in user_data["friends"]: # reaching into the list of dictionaries
+        for item in movie["watched"]:
+            friends_movies.append(item) # this is now a list of the friend's movies
+
+    for item in friends_movies: # we are accesing the list we created of movies watched by friends
+        if item not in user_movies: # it the friends movie is not in the movies the user has seen
+            if item not in unique: # if that movie is also not in our list of unique movies, we add to unique
+                if item["host"] in user_data["subscriptions"]: # if streaming service is in subscription
+                    unique.append(item) # if streaming service is found, add to unique
+
+    return unique
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    user_movies = []
+    friends_movies = []
+    most_genre = []
+    recommended_movies = []
+
+    for movie in user_data["watched"]: # this is to help create the user watch list
+        user_movies.append(movie) #user_movies has a list of the movies the user has seen
+        most_genre.append(movie["genre"]) # to get to the genre
+
+    if len(user_movies) == 0:
+        return friends_movies # if the list of user is completely empty
+
+    max(set(most_genre), key=most_genre.count) # will get the most watched genre, will 

```

Comment:
This is neat and works! That being said, I'd encourage you as we're learning to also be thinking about how to do things in the less fancy way.